engine/gui/src/gui.cpp

```diff
@@ -1306,41 +1291,24 @@ Result DeleteDynamicTexture(HScene scene, const dmhash_t texture_hash)
                 if (n->m_ClipperIndex != INVALID_INDEX) {
                     InternalClippingNode& clipper = clippers[n->m_ClipperIndex];
                     if (clipper.m_NodeIndex == index) {
-                        bool root_clipper = scope == 0x0;
-                        Scope tmp_scope(0, order);
-                        Scope* current_scope = scope;
-                        if (current_scope == 0x0) {
-                            current_scope = &tmp_scope;
-                            ++order;
-                        } else {
-                            Increment(current_scope);
-                        }
-                        uint64_t clipping_key = CalcRenderKey(current_scope, 0, 0);
-                        uint64_t render_key = CalcRenderKey(current_scope, layer, 1);
-                        CollectRenderEntries(scene, n->m_ChildHead, 2, current_scope, clippers, render_entries);
-                        if (layer > 0) {
-                            render_key = CalcRenderKey(current_scope, layer, 1);
-                        }
+                        uint64_t clipping_key = CalcRenderKey(layer, order++);
+                        uint64_t render_key   = CalcRenderKey(layer, order++);
+
+                        order = CollectRenderEntries(scene, n->m_ChildHead, order, clippers, render_entries);
```

Comment:
I do like the cleanup!
A bit hard to reason about though but the recursive step seems ok I guess.
Also, don't forget to update the documentation!